src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -173,6 +175,18 @@ public void run() {
                     }
                 } catch (InvalidProtocolBufferException ipbe) {
                     LOG.debug("Error while processing protobuf: {}", ipbe.getLocalizedMessage());
+                } catch (JsonParseException jpe) {
+                    LOG.warn("Error while processing protobuf: {}", jpe.getLocalizedMessage());
+                } catch (SocketException se) {
+                    LOG.warn("Socket error, will reconnect: {}", se.getLocalizedMessage());
+                    LOG.debug("StackTrace", se);
+                    try {
+                        close();
```

Comment:
Basically I want reconnection to happen only when `autoReconnect` property is set to `true` in `ChromeCast` class. From my opinion here we should do the following:

1) Introduce `autoReconnect` property in the `Channel` class, which will reflect all changes from the `ChromeCast` class, and the default value will be `true`.

2) When `SocketException` is caught - always `close()` channel, as it is now in your implementation. Then if  `autoReconnect` property is set to `true` - call the `connect()` method, do nothing otherwise.

I think this change should be quite simple and will leave the reconnection logic consistent.